src/utils/convertPipes/index.js

```diff
@@ -20,68 +28,150 @@ const getAnswer = (ctx, answerId) =>
     .filter(answer => isPipeableType(answer))
     .find(answer => answer.id === answerId);
 
+const DATE_FORMAT_MAP = {
+  "dd/mm/yyyy": "d MMMM yyyy",
+  "mm/yyyy": "MMMM yyyy",
+  yyyy: "yyyy"
+};
+
+const TRANSFORM_MAP = {
+  Currency: { format: FORMAT_CURRENCY, transformKey: NUMBER_TRANSFORMATION },
+  Number: { format: FORMAT_NUMBER, transformKey: NUMBER_TRANSFORMATION },
+  Date: { format: FORMAT_DATE, transformKey: DATE_TRANSFORMATION },
+  DateRange: { format: FORMAT_DATE, transformKey: DATE_TRANSFORMATION }
+};
+
+const transform = (dataType, value) => ({
+  value,
+  format: TRANSFORM_MAP[dataType].format,
+  options: (source, identifier, dateFormat) => ({
+    [TRANSFORM_MAP[dataType].transformKey]: {
+      source,
+      identifier
+    },
+    ...(dataType === "Date" && {
+      date_format:
+        DATE_FORMAT_MAP[dateFormat] ||
+        (source === "metadata" && DATE_FORMAT_MAP["dd/mm/yyyy"])
+    })
+  })
+});
```

Comment:
Could we turn this into an explicitly returned function and try to make it a bit more readable? I'm imagining something like

```
const transform = (dataType, value) => {
    let dateFormat;
    if(dataType === "Date"){
        . . .
    }
    const options = . . .
    
    return {
        value, 
        format, 
        options, 
        date_format: dateFormat
}
```

Just a small suggestion, could we rename `dataType` to `answerType`? When I first read through it I thought it was variable data type not Author data type. But that might just be me.